pipeline/merger/src/main/scala/uk/ac/wellcome/platform/merger/services/RecorderPlaybackService.scala

```diff
@@ -29,8 +31,7 @@ class RecorderPlaybackService(
   def fetchAllWorks(workIdentifiers: Seq[WorkIdentifier])
     : Future[Seq[Option[TransformedBaseWork]]] = {
     Future.sequence(
-      workIdentifiers
-        .map { getWorkForIdentifier }
+      workIdentifiers.map(id => Future.fromTry(Try(getWorkForIdentifier(id))))
```

Comment:
Maybe drop the intermediate `Try` here?

```suggestion
      workIdentifiers.map(id => Future { getWorkForIdentifier(id) }))
```

Reply:
The docs for scala `Future` say that `fromTry`

> Creates an already completed Future with the specified result or exception.

whereas `apply`

> Starts an asynchronous computation and returns a Future instance with the result of that computation.

so these two are a little different. You are probably right, I guess I wrapped in a try as wanted to execute right away rather than in a separate worker, but I guess I shouldn't care :thinking: